packages/workspace/src/command-line/workspace-schematic.ts

```diff
@@ -98,29 +155,42 @@ function compileToolsDir(outDir: string) {
 }
 
 function constructCollection() {
-  const schematics = {};
-  fs.readdirSync(schematicsDir()).forEach((c) => {
-    const childDir = path.join(schematicsDir(), c);
-    if (exists(path.join(childDir, 'schema.json'))) {
-      schematics[c] = {
-        factory: `./${c}`,
-        schema: `./${path.join(c, 'schema.json')}`,
-        description: `Schematic ${c}`,
+  const schematics = fs
+    .readdirSync(schematicsDir())
+    .reduce((acc, schematicName) => {
+      const childDir = path.join(schematicsDir(), schematicName);
+      const schemaJsonPath = path.join(childDir, 'schema.json');
+      if (!exists(schemaJsonPath)) {
+        return acc;
+      }
+
+      const schematicMetadata = {
+        factory: `./${schematicName}`,
+        schema: `./${path.join(schematicName, 'schema.json')}`,
+        description:
+          getSchemaJson(schemaJsonPath).description ||
+          `Schematic ${schematicName}`,
       };
-    }
-  });
+      acc[schematicName] = schematicMetadata;
+
+      return acc;
+    }, {} as Schema['schematics']);
+
   return {
-    name: 'workspace-schematics',
+    name: collectionName.replace('.json', ''),
     version: '1.0',
     schematics,
   };
 }
 
-function saveCollection(dir: string, collection: any) {
-  writeFileSync(
-    path.join(dir, 'workspace-schematics.json'),
-    JSON.stringify(collection)
-  );
+function getSchemaJson(schemaPath: string): SchematicSchema {
+  const schema = readJsonFile<SchematicSchema>(schemaPath);
```

Comment:
inline variable

Reply:
done, I'd suggest to enable proper eslint rules for that (sonarjs has such a rule)